File: python/251-300/252. Meeting Rooms.py
# Definition for an interval.
# class Interval(object):
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution(object):
    def canAttendMeetings(self, intervals):
        """
        :type intervals: List[Interval]
        :rtype: bool
        """
        intervals.sort(key = lambda interval:interval.start)
        for i in range(len(intervals) - 1):
            if intervals[i+1].start < intervals[i].end:
                return False
        return True

File: cpp/251-300/252. Meeting Rooms.cc
/**
 * Definition for an interval.
 * struct Interval {
 *     int start;
 *     int end;
 *     Interval() : start(0), end(0) {}
 *     Interval(int s, int e) : start(s), end(e) {}
 * };
 */

class Solution {
public:
    static bool compare(Interval l1, Interval l2) {
        return (l1.start < l2.start);
    }
    bool canAttendMeetings(vector<Interval>& intervals) {
        if (intervals.empty())
            return true;
        sort(intervals.begin(), intervals.end(), compare);
        for (int i = 1; i < intervals.size(); i++) {
            if (intervals[i].start < intervals[i - 1].end)
                return false;
        }
        return true;
    }
};